validation/superchain-version_test.go

```diff
@@ -13,12 +14,45 @@ import (
 	"github.com/ethereum-optimism/superchain-registry/validation/standard"
 	"github.com/stretchr/testify/require"
 
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/crypto"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/ethclient"
 )
 
+// list of contracts to check for version/bytecode uniformity
+// TODO - The list intentionally omits contracts which have immutables because the bytecode check needs
+// to be enhanced to mask the immutable values in the bytecode (contracts related to fault proofs) so that any check against a hash
+// of the bytecode can validate what's on-chain.
+var (
+	contractsToCheck = []string{
+		"L1CrossDomainMessengerProxy",
+		"L1ERC721BridgeProxy",
+		"L1StandardBridgeProxy",
+		"OptimismMintableERC20FactoryProxy",
+		"OptimismPortalProxy",
+		"SystemConfigProxy",
+		"AnchorStateRegistryProxy",
+		"DelayedWETHProxy",
+		"DisputeGameFactoryProxy",
+		"FaultDisputeGame",
+		"MIPS",
+		"PermissionedDisputeGame",
+		"PreimageOracle",
+	}
+
+	contractsToSkip = []string{
+		"AnchorStateRegistryProxy",
+		"DelayedWETHProxy",
+		"DisputeGameFactoryProxy",
+		"FaultDisputeGame",
+		"MIPS",
+	}
```

Comment:
Ideally these contracts are declared to be skipped _only_ in the scope of the bytecode hash check, and not in the scope of the version check.